service/service.py

```diff
@@ -211,6 +223,42 @@ def delete_inventory(product_id, condition):
     app.logger.info("Inventory with product_id %d and condition %s deleted", product_id, condition)
     return make_response("", status.HTTP_204_NO_CONTENT)
 
+
+################################################################################
+# UPDATE AN EXISTING PRODUCT's QUANTITY 
+################################################################################
+@app.route("/inventory/<int:product_id>/<string:condition>/<string:operation>/<int:amount>", methods=["PUT"])
+def update_stock(product_id, condition, operation, amount):
+    """Updates the inventory with the given product_id and condition"""
+    app.logger.info("Request to update quantity of product in inventory with product_id %d and condition %s", product_id, condition)
+    
+
+    if amount == 0:
+        return bad_request("Wrong update amount parameter specified . Amount can only be a non zero whole number Eg : /inventory/123/new/add/1")
+
+    if operation != "add" and operation != "sub":
+        return bad_request("Wrong operation specified. Operation can only be add or sub in http request. Eg : /inventory/123/new/add/1")
+
+    inventory = Inventory.find(product_id, condition)
+    if not inventory:
+        raise NotFound("Inventory with product_id {} and condition {} was not found.".format(product_id, condition))
+    
+    msg_tmp = " "
+
+    if operation == "add":
+        inventory.quantity = inventory.quantity + amount
+        msg_tmp = "Added "
+    else:
+        inventory.quantity = inventory.quantity - amount
```

Comment:
Can you add another logic after this statement? When the ```quantity=0``` then it should be unavailable.
```
if inventory.quantity==0:
    inventory.available=0
```

Reply:
added